### PR TITLE
CCIP-3591 USDC Integration Tests

### PR DESCRIPTION
**Pull request by mateusz-sekara**

### CODE REVIEW

File: deployment/ccip/changeset/active_candidate_test.go [hunk 1]
```diff
@@ -152,15 +152,7 @@ func TestActiveCandidate(t *testing.T) {
...
-	ocr3ConfigMap, err := ccdeploy.BuildOCR3ConfigForCCIPHome(
-		deployment.XXXGenerateTestOCRSecrets(),
-		state.Chains[destCS].OffRamp,
-		e.Chains[destCS],
-		destCS,
-		tokenConfig.GetTokenInfo(e.Logger, state.Chains[destCS].LinkToken, state.Chains[destCS].Weth9),
-		nodes.NonBootstraps(),
-		rmnHomeAddress,
-	)
+	ocr3ConfigMap, err := ccdeploy.BuildOCR3ConfigForCCIPHome(deployment.XXXGenerateTestOCRSecrets(), state.Chains[destCS].OffRamp, e.Chains[destCS], destCS, tokenConfig.GetTokenInfo(e.Logger, state.Chains[destCS].LinkToken, state.Chains[destCS].Weth9), nodes.NonBootstraps(), rmnHomeAddress, nil)
```
Comment (makramkd) [inline deployment/ccip/changeset/active_candidate_test.go]: Isn't this being on one line less readable?

File: deployment/ccip/test_helpers.go [hunk 2]
```diff
@@ -525,18 +526,19 @@ func DeployTransferableToken(
 	}
 
 	// Add burn/mint permissions
-	if err := grantMintBurnPermissions(chains[src], srcToken, srcPool.Address()); err != nil {
+	if err := grantMintBurnPermissions(nil, chains[src], srcToken, srcPool.Address()); err != nil {
```
Comment (makramkd) [inline deployment/ccip/test_helpers.go]: Passing in nil as the logger doesn't cause panics? :O

File: deployment/ccip/test_helpers.go [hunk 3]
```diff
@@ -3,13 +3,13 @@ package ccipdeployment
 import (
 	"context"
 	"fmt"
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/reader"
```
Comment (makramkd) [inline deployment/ccip/test_helpers.go]: need to run goimports here. Are we running golangci-lint on this deployment mod?

File: deployment/ccip/test_usdc_helpers.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+func ConfigureUSDCTokenPools(lggr logger.Logger, chains map[uint64]deployment.Chain, src, dst uint64, state CCIPOnChainState) (*burn_mint_erc677.BurnMintERC677, *burn_mint_erc677.BurnMintERC677, error) {
+	srcToken := state.Chains[src].BurnMintTokens677[USDCSymbol]
+	dstToken := state.Chains[dst].BurnMintTokens677[USDCSymbol]
+	srcPool := state.Chains[src].USDCTokenPool
+	dstPool := state.Chains[dst].USDCTokenPool
+
+	// Attach token pools to registry
+	if err := attachTokenToTheRegistry(chains[src], state.Chains[src], chains[src].DeployerKey, srcToken.Address(), srcPool.Address()); err != nil {
+		return nil, nil, err
```
Comment (makramkd) [inline deployment/ccip/test_usdc_helpers.go:30]: nit for this return and those below: does the `err` have enough context to debug or should we wrap with `fmt.Errorf`?

File: deployment/ccip/test_usdc_helpers.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+			)
+			return ContractDeploy[*mock_usdc_token_transmitter.MockE2EUSDCTransmitter]{
+				transmitterAddress, transmitterContract, tx, deployment.NewTypeAndVersion(USDCMockTransmitter, deployment.Version1_0_0), err2,
+			}
+		})
+	if err != nil {
+		lggr.Errorw("Failed to deploy mock USDC transmitter", "err", err)
+		return nil, nil, nil, nil, err
+	}
+
```
Comment (makramkd) [inline deployment/ccip/test_usdc_helpers.go:182]: Doesn't seem like these tx's (below this point, above I see one) are `chain.Confirm`d, do they need to be?

File: integration-tests/smoke/ccip_test.go [hunk 6]
```diff
@@ -1,11 +1,10 @@
 package smoke
 
 import (
-	"math/big"
-	"testing"
-
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
```
Comment (makramkd) [inline integration-tests/smoke/ccip_test.go]: Bad import reorder

File: integration-tests/smoke/ccip_usdc_test.go [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	commonconfig "github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/shared/generated/burn_mint_erc677"
+	"golang.org/x/exp/maps"
+	"math/big"
```
Comment (makramkd) [inline integration-tests/smoke/ccip_usdc_test.go]: bad import ordering

File: integration-tests/smoke/ccip_usdc_test.go [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+	require.NoError(t, err)
+
+	allChainSelectors := maps.Keys(e.Chains)
+	sourceChain := allChainSelectors[0]
+	destChain := allChainSelectors[1]
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		response := `{
+			"status": "complete",
+			"attestation": "0x9049623e91719ef2aa63c55f357be2529b0e7122ae552c18aff8db58b4633c4d3920ff03d3a6d1ddf11f06bf64d7fd60d45447ac81f527ba628877dc5ca759651b08ffae25a6d3b1411749765244f0a1c131cbfe04430d687a2e12fd9d2e6dc08e118ad95d94ad832332cf3c4f7a4f3da0baa803b7be024b02db81951c0f0714de1b"
```
Comment (makramkd) [inline integration-tests/smoke/ccip_usdc_test.go]: How did we come up with this attestation? Comment/instructions to generate would be useful

File: integration-tests/smoke/ccip_usdc_test.go [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccdeploy "github.com/smartcontractkit/chainlink/deployment/ccip"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func TestUSDCTokenTransfer(t *testing.T) {
```
Comment (makramkd) [inline integration-tests/smoke/ccip_usdc_test.go:34]: Is this test running in CI? I see that its in a different file, if its not in ccip_test.go it won't automatically run, you'll have to add a new section in the e2e-tests.yaml or whatever the file is called

File: integration-tests/smoke/ccip_usdc_test.go (hunk 8 above)
Comment (mateusz-sekara) [reply integration-tests/smoke/ccip_usdc_test.go]: I will clarify that, but it's random, we don't verify of the attestation payload in these tests

File: deployment/ccip/test_usdc_helpers.go (hunk 5 above)
Comment (mateusz-sekara) [reply deployment/ccip/test_usdc_helpers.go:182]: DeployContract doesn't need to be confirmed, at least this is how it works for other contract deployment

File: integration-tests/smoke/ccip_usdc_test.go (hunk 9 above)
Comment (mateusz-sekara) [reply integration-tests/smoke/ccip_usdc_test.go:34]: Addec config

**Review by makramkd — APPROVED** [review]

**Review by lukaszcl — APPROVED** [review]

**Review by b-gopalswami — APPROVED** [review]

**Pull request merged**